ksp_plugin/flight_plan_optimizer.hpp

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include "base/not_null.hpp"
+#include "geometry/instant.hpp"
+#include "geometry/space.hpp"
+#include "ksp_plugin/celestial.hpp"
+#include "ksp_plugin/flight_plan.hpp"
+#include "ksp_plugin/frames.hpp"
+#include "numerics/fixed_arrays.hpp"
+#include "numerics/gradient_descent.hpp"
+#include "physics/reference_frame.hpp"
+#include "quantities/named_quantities.hpp"
+#include "quantities/quantities.hpp"
+
+namespace principia {
+namespace ksp_plugin {
+namespace _flight_plan_optimizer {
+namespace internal {
+
+using namespace principia::base::_not_null;
+using namespace principia::geometry::_instant;
+using namespace principia::geometry::_space;
+using namespace principia::ksp_plugin::_celestial;
+using namespace principia::ksp_plugin::_flight_plan;
+using namespace principia::ksp_plugin::_frames;
+using namespace principia::numerics::_fixed_arrays;
+using namespace principia::numerics::_gradient_descent;
+using namespace principia::physics::_reference_frame;
+using namespace principia::quantities::_named_quantities;
+using namespace principia::quantities::_quantities;
+
+// A class to optimize a flight to go through or near a celestial.
+class FlightPlanOptimizer {
+ public:
+  // Constructs an optimizer for |flight_plan|.  |flight_plan| must overlive
```

Comment:
The word _overlive_ is unattested in our codebase, whereas _outlive_ is attested: https://github.com/search?q=repo%3Amockingbirdnest%2FPrincipia+outlive&type=code.